core/src/test/scala/com/rackspace/com/papi/components/checker/GivenAWadlWithMetaTransforms.scala

```diff
@@ -0,0 +1,279 @@
+      it should behave like accessIsAllowed(validator, methodGet, s"/custom/metadata/${target}foo", roles, description)
+    }
+    it should behave like accessIsAllowedWhenNoXRoles(validator, methodGet, s"/custom/metadata/${target}foo", description)
+  }
+
+  // Custom PUT's and DELETE's are allowed or forbidden based on roles.
+// @TODO: Original double loop with PUT's and DELETE's for Custom targets.
+//  List(methodPut, methodDelete).foreach { method =>
+//    targetsCustom.foreach { target =>
+//      rolesList.foreach { roles =>
+//        // IF the list of roles contains the target
+//        // OR the list of roles contains admin OR superAdmin,
+//        // THEN access should be allowed;
+//        // ELSE IF the target is the overloaded target
+//        // AND the list of roles contains the overloaded role,
+//        // THEN access should be allowed;
+//        // ELSE access should be forbidden.
+//        if (targetInRoles(target, roles, List("admin", "superAdmin"))) {
+//          it should behave like accessIsAllowedWithContent(validator, method, s"/custom/metadata/${target}foo", roles, putContent, description)
+//        } else if (target.equals("service:") && targetInRoles("another_role", roles, List.empty)) {
+//          it should behave like accessIsAllowedWithContent(validator, method, s"/custom/metadata/${target}foo", roles, putContent, description)
+//        } else {
+//          it should behave like accessIsForbidden(validator, method, s"/custom/metadata/${target}foo", roles, description)
+//        }
+//      }
+//      it should behave like accessIsForbiddenWhenNoXRoles(validator, method, s"/custom/metadata/${target}foo", description)
+//    }
+//  }
+
+  targetsCustom.foreach { target =>
+    rolesList.foreach { roles =>
+      ////////////////////////////////////////
+      // IF the list of roles contains the target
+      // OR the list of roles contains admin OR superAdmin,
+      // THEN access should be allowed;
+      // ELSE IF the target is the overloaded target
+      // AND the list of roles contains the overloaded role,
+      // THEN access should be allowed;
+      // ELSE access should be forbidden.
+      ////////////////////////////////////////
+      // @TODO: Huh??? Why doesn't this work?
+      //if (targetInRoles(target, roles, List("admin", "superAdmin"))) {
+      //  it should behave like accessIsAllowed(validator, methodPut, s"/custom/metadata/${target}foo", roles, description)
+      //} else if (target.equals("service:") && targetInRoles("another_role", roles, List.empty)) {
+      //  it should behave like accessIsAllowed(validator, methodPut, s"/custom/metadata/${target}foo", roles, putContent, description)
+      ////////////////////////////////////////
+      // IF the list of roles contains admin OR superAdmin,
+      // THEN access should be allowed;
+      // ELSE access should be forbidden.
+      if (targetInRoles("", roles, List("admin", "superAdmin"))) {
```

Comment:
Yea, admins are supposed to have access to PUT and DELETE on metadata and metadata/{key}.  But which roles to use for admins depends on what the metadata extension.  Basically any role that uses a pattern of "*" is considered an admin.